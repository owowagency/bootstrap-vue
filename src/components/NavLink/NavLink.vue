<template>
    <a
        class="nav-link"
        :class="classes"
        href="#"
    >
        <!-- @slot Displays the content of the nav link -->
        <slot />
    </a>
</template>

<script lang="ts">
import useActive, {activeProps} from '@/composables/useActive';
import useDisabled, {disabledProps} from '@/composables/useDisabled';
import {computed} from 'vue';
import useClasses from '@/composables/useClasses';
</script>

<script lang="ts" setup>
const props = defineProps({
    ...activeProps,
    ...disabledProps,
});

const {classes} = useClasses(computed(() => [
    useActive(props.active).activeClass.value,
    useDisabled(props.disabled).disabledClass.value,
]));
</script>

<docs>
```vue
<NavLink href="https://github.com/owowagency/bootstrap-vue">
    GitHub
</NavLink>
```


## Imported Props

| Prop name | Description             | Type    | Values | Default   |
| --------- | ----------------------- | ------- | ------ | --------- |
| active    | Adds the `active` class | boolean | -      | false     |
| disabled  | Disables the component  | boolean | -      | false     |
</docs>

