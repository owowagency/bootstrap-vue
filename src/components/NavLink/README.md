# NavLink

```vue
<NavLink href="https://github.com/owowagency/bootstrap-vue">
    GitHub
</NavLink>
```

## Imported Props

| Prop name | Description             | Type    | Values | Default |
| --------- | ----------------------- | ------- | ------ | ------- |
| active    | Adds the `active` class | boolean | -      | false   |
| disabled  | Disables the component  | boolean | -      | false   |

## Slots

| Name    | Description                          | Bindings |
| ------- | ------------------------------------ | -------- |
| default | Displays the content of the nav link |          |
