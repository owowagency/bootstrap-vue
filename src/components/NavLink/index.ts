import NavLink from './NavLink.vue';

export default NavLink;
