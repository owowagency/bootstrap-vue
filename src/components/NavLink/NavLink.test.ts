import NavLink from '.';
import {describe} from 'vitest';

describe('template', () => {
    componentRenderTest(NavLink);

    componentSlotRenderTest(NavLink);

    componentWrapperClassTest(NavLink, {active: true}, 'active');

    componentWrapperClassTest(NavLink, {disabled: true}, 'disabled');
});
