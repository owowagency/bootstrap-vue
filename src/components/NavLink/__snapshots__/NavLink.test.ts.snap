// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<a class=\\"nav-link\\" href=\\"#\\">
  <!-- @slot Displays the content of the nav link -->
</a>"
`;
